service-discovery/stork-custom/src/test/java/io/quarkus/ts/stork/custom/StorkCustomLoadBalancerIT.java

```diff
@@ -0,0 +1,86 @@
+package io.quarkus.ts.stork.custom;
+
+import static org.hamcrest.CoreMatchers.hasItems;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+import static org.hamcrest.Matchers.is;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.http.HttpStatus;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.bootstrap.RestService;
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.test.services.QuarkusApplication;
+import io.restassured.response.ValidatableResponse;
+
+import junit.framework.AssertionFailedError;
+
+@QuarkusScenario
+public class StorkCustomLoadBalancerIT {
+
+    public static final String PREFIX = "ping-";
+    private static final String DEFAULT_PONG_REPLICA_RESPONSE = "pongReplica";
+    private static final String DEFAULT_PONG_RESPONSE = "pong";
+    private static final String PONG_PORT = getAvailablePort();
+    private static final String PONG_REPLICA_PORT = getAvailablePort();
+
+    @QuarkusApplication(classes = PongResource.class)
+    static RestService pongService = new RestService().withProperty("quarkus.http.port", PONG_PORT);
+
+    @QuarkusApplication(classes = PongReplicaResource.class)
+    static RestService pongReplicaService = new RestService().withProperty("quarkus.http.port", PONG_REPLICA_PORT);
+
+    @QuarkusApplication
+    static RestService mainPingService = new RestService()
+            .withProperty("stork.pong.load-balancer", "simple")
+            .withProperty("stork.pong.service-discovery", "simple")
+            .withProperty("stork.pong-replica.load-balancer", "simple")
+            .withProperty("stork.pong-replica.service-discovery", "simple")
+            .withProperty("stork.pong.service-discovery.pongServicePort", () -> "" + pongService.getPort())
+            .withProperty("stork.pong-replica.service-discovery.pongReplicaServicePort",
+                    () -> "" + pongReplicaService.getPort());
+
+    @Test
+    public void storkCustomLoadBalancer() {
```

Comment:
It also tests the custom service discovery, so maybe we should reflect that in the names of the method and the class.